lib/src/hitmap.dart

```diff
@@ -35,15 +35,21 @@ class HitMap {
 
   /// Creates a single hitmap from a raw json object.
   ///
+  /// Note that when [checkIgnoredLines] is `true` all files will be
+  /// read to get ignore comments. This will add some overhead.
+  /// To combat this when calling this function multiple times from the
+  /// same source (e.g. test runs of different files) a cache is taken
+  /// via [ignoredLinesInFilesCache]. If this cache contains the parsed
+  /// data for the specific file already, the file will not be read and
+  /// parsed again.
+  ///
   /// Throws away all entries that are not resolvable.
-  static Future<Map<String, HitMap>> parseJson(
+  static Map<String, HitMap> parseJsonSync(
     List<Map<String, dynamic>> jsonResult, {
-    bool checkIgnoredLines = false,
-    @Deprecated('Use packagePath') String? packagesPath,
-    String? packagePath,
-  }) async {
-    final resolver = await Resolver.create(
-        packagesPath: packagesPath, packagePath: packagePath);
+    required bool checkIgnoredLines,
+    required Map<String, List<List<int>>?> ignoredLinesInFilesCache,
```

Comment:
Is there ever a case where we wouldn't want this caching, or would want more than one cache? I wonder if it would be cleaner to just make the cache a static variable, rather than an argument to this function.

Reply:
If I understand you correctly in "a static variable" that's basically never a good idea. For one thing it leaks unless explicitly cleared, additionally, if a long-running process calls several times with time between calls (e.g. between hot reloads) the cache will no longer be valid.

For not having the cache --- if it's just a one-shot thing (i.e. it's not called several times for the same files as in the flutter case) one wouldn't need the cache.

As for more than one cache, that would for instance be the case when asking at two different points in time where the files have changed.